pyrokinetics/equilibrium.py

```diff
@@ -159,6 +162,203 @@ def read_geqdsk(self, psi_n_lcfs=1.0):
 
         self.R_major = InterpolatedUnivariateSpline(psi_n, R_major)
 
+    def read_transp_cdf(
+        self, time=-1, nr=None, nz=None, Rmin=None, Rmax=None, Zmin=None, Zmax=None
+    ):
+        """
+
+        Read in TRANSP netCDF and populates Equilibrium object
+
+        """
+
+        import netCDF4 as nc
+        import numpy as np
+        from scipy.interpolate import (
+            RBFInterpolator,
+            InterpolatedUnivariateSpline,
+            RectBivariateSpline,
+        )
+
+        data = nc.Dataset(self.eq_file)
+
+        nradial = len(data["XB"][-1, :])
+
+        time_cdf = data["TIME3"][:]
+
+        if isinstance(time, int):
+            itime = time
+        elif isinstance(time, float):
+            itime = np.argmin(abs(time_cdf - time))
```

Comment:
This uses `time` in two different ways depending on its type, which feels a little error prone. Instead, perhaps add a `time_index: int = -1` input alongside `time: float = None`, and then:

```python
if time_index != -1 and time is not None:
    raise ValueError("Cannot set both `time` and `time_index`")

if time is not None:
    time_index = np.argmin(np.abs(time_cdf - time))
```

The alternative is make sure the behaviour is very well documented.